[Merged by Bors] - Add cmdline flag for tcp interface

lrettig opened the pull request:
This, in combination with --acquire-port=false, prevents the obnoxious
warning popup in macOS.

## Motivation
Fixes the issue described in https://github.com/spacemeshos/go-spacemesh/issues/289#issuecomment-574906535 where macOS prompts the user to allow incoming connections every time the node is run. While this permission is cached by the OS for a production app, there is no way around it in development since it reappears each time the app is recompiled.

To prevent this popup, with this change, run go-spacemesh with `--acquire-port=false --tcp-interface 127.0.0.1` (a similar flag was added for the new GRPC server in a10ce1e146367942e275a9225eef1fee6f6e38f3).

## Changes
- adds a new commandline flag
- removes the hardcoded "0.0.0.0" from P2P listener

## Test Plan
- Run node with and without the new flag and check if it's listening on the right interface

## TODO
- [ ] Add test

lrettig committed:
Add cmdline flag for tcp interface

This, in combination with --acquire-port=false, prevents the obnoxious
warning popup in macOS.

lrettig committed:
Make linter happy

y0sher commented on cmd/root.go:
```diff
@@ -68,7 +68,9 @@ func AddCommands(cmd *cobra.Command) {
-		config.P2P.TCPPort, "TCP Port to listen on")
+		config.P2P.TCPPort, "inet port for P2P listener")
+	cmd.PersistentFlags().StringVar(&config.P2P.TCPInterface, "tcp-interface",
+		config.P2P.TCPInterface, "inet interface for P2P listener")
```
this is the iface inet address right? I mean the iface name is probably "lo".

lrettig replied:
It expects an IP address here, not an interface name, I think. Will double check this and make it clearer in the description if it must be an IP.

y0sher reviewed (approved):
looks good overall added a small comment about naming..

noamnelke commented:
Regarding `--acquire-port=false` - this makes total sense in development, when you're not actually connecting to a real network. Ideally it shouldn't be something you do manually, but part of the default development config.

It's important that it's enabled by default in real-world use, as it decreases the number of non-routable nodes (by using UPnP to automatically forward ports on the local network).

Do you get this prompt in tests (I never do)? Or do you actually run a node binary on your machine?

lrettig commented:
> Do you get this prompt in tests (I never do)? Or do you actually run a node binary on your machine?

I get this prompt many many times during tests. Every new test that tries to listen on tcp causes it to popup again. I also get it when running the binary locally.

noamnelke commented:
Are we merging this?

lrettig commented:
> Are we merging this?

Just need to add a test, will do asap

lrettig committed:
Prevent a null pointer exception

Don't try to dial if the listen address isn't set first.
Clean up some comments and error messages.

lrettig committed:
Add test for TCPInterface

Also make commandline arg description clearer

lrettig committed:
Make commandline arg description clearer

lrettig commented:
bors merge

lrettig committed:
Update test too

lrettig commented:
bors merge
